app/policies/plan_policy.rb

```diff
@@ -9,47 +9,47 @@ def initialize(user, plan)
   end
 
   def show?
-    @plan.readable_by?(@user.id)
+    @plan.readable_by?(@user.id) && Role.find_by(user_id: @user.id, plan_id: @plan.id).active
```

Comment:
The check is probably not needed, but If a user removes themself from a plan, and other users cant see that they can access the plan, I like ensuring that behavior in code.